src/mmw/js/src/modeling/models.js

```diff
@@ -108,16 +108,23 @@ var ProjectModel = Backbone.Model.extend({
         this.set('name', newName);
     },
 
+    // Flag to prevent double POSTing of a project.
+    saveCalled: false,
+
     saveProjectAndScenarios: function() {
         if (!App.user.loggedInUserMatch(this.get('user_id'))) {
+            // Fail fast if the user can't save the project.
             return;
         }
 
-        if (!this.get('id')) {
+        if (!this.get('id') && this.saveCalled) {
+            // Fail fast if we are in the middle of our first save.
+            return;
+        } else if (!this.get('id') && !this.saveCalled) {
             // We haven't saved the project before, save the project and then
-            // set the project ID on each scenario, then reattach the senarios
-            // to the project.
+            // set the project ID on each scenario.
             var self = this;
+            this.saveCalled = true;
```

Comment:
A more direct question: are we only expecting to use `saveProjectAndScenarios` once per session?